src/compiler.zig

```diff
@@ -684,10 +685,46 @@ pub const Compiler = struct {
             .{},
         ));
 
-        const locals = self.scope.symbols.items().len;
+        // get locals
+        var locals = try std.ArrayList(*lir.Inst).initCapacity(self.ir.gpa, self.scope.symbols.items().len);
```

Comment:
The result is probably the same since you are using `appendAssumeCapacity` but you could `try self.ir.gpa.alloc(*lir.Inst, self.scope.symbols.items().len);` and use a slice instead for a little less overhead.

Reply:
Great idea! Can't believe I didn't think of it. I do it at function arguments as well. Thanks Alex! :)